packages/botonic-react/src/components/carousel.jsx

```diff
@@ -35,14 +57,81 @@ export const Carousel = props => {
     ...{ enable: true, autoHide: true },
     ...getThemeProperty('scrollbar'),
   }
+  const [hasLeftArrow, setLeftArrow] = useState(false)
+  const [hasRightArrow, setRightArrow] = useState(true)
+  const carouselRef = useRef(null)
+  const CustomCarouselLeftArrow = getThemeProperty(
+    'carousel.arrow.left',
+    undefined
+  )
+  const CustomCarouselRightArrow = getThemeProperty(
+    'carousel.arrow.right',
+    undefined
+  )
+  const enableCarouselArrows = getThemeProperty('carousel.enableArrows', true)
+
+  const scrollCarouselBy = value => {
+    carouselRef.current.scrollBy({
+      left: value,
+      behavior: 'smooth',
+    })
+  }
+
+  const setArrowsVisibility = event => {
+    const carousel = event.currentTarget
+    carousel.scrollLeft == 0 ? setLeftArrow(false) : setLeftArrow(true)
+    carousel.scrollLeft == carousel.scrollWidth - carousel.offsetWidth
+      ? setRightArrow(false)
+      : setRightArrow(true)
+  }
+
+  const getArrows = () => {
+    return (
+      <>
+        {hasLeftArrow &&
+          (CustomCarouselLeftArrow ? (
+            <CustomCarouselLeftArrow scrollCarouselBy={scrollCarouselBy} />
+          ) : (
+            <StyledArrowContainer
+              style={{ left: 0 }}
```

Comment:
the trick to not use inline styles with `styled-components` is to pass a prop like:
`left={0}`
and then you can use them inside the `StyledArrowContainer` like:
```
const StyledArrowContainer = styled.div`
  position: absolute;
  left: ${props => props.left}px;
  right: ${props => props.right}px;
```